services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/PersistenceOperationsBolt.java

```diff
@@ -51,9 +55,16 @@ protected void handleInput(Tuple input) {
         final String correlationId = input.getStringByField("correlationId");
         log.debug("Received operation request");
 
-        List<InfoData> result = processRequest(input, request);
-        getOutput().emit(input, new Values(result, correlationId));
+        try {
+            List<InfoData> result = processRequest(input, request);
+            getOutput().emit(input, new Values(result, correlationId));
+        } catch (IslNotFoundException e) {
+            ErrorData data = new ErrorData(ErrorType.NOT_FOUND,
+                    e.getMessage(),
+                    "ISL does not exist.");
+            getOutput().emit(StreamType.ERROR.toString(), input, new Values(data, correlationId));
+        }
     }
 
-    abstract List<InfoData> processRequest(Tuple tuple, BaseRequest request);
+    abstract List<InfoData> processRequest(Tuple tuple, BaseRequest request) throws IslNotFoundException;
```

Comment:
It's a little bit confusing that you throw concrete (ISL related) exception in abstract method

Reply:
Fixed.